app/acapy.js

```diff
@@ -0,0 +1,86 @@
+const { ACAPY_API_KEY, ACAPY_ADMIN_SERVICE } = require('./env')
+
+const querySubWallets = async (subwalletName) => {
+  const fetch = (await import('node-fetch')).default
+
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}/multitenancy/wallets`)
+  const search = new URLSearchParams({ wallet_name: subwalletName })
+  url.search = search.toString()
+
+  const queryResponse = await fetch(url, {
+    headers: {
+      'x-api-key': ACAPY_API_KEY,
+    },
+  })
+
+  const wallets = await queryResponse.json()
+
+  return wallets.results
+}
+
+const createSubWallet = async (body) => {
+  const fetch = (await import('node-fetch')).default
+
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}/multitenancy/wallet`)
+  const createResponse = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'x-api-key': ACAPY_API_KEY,
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify(body),
+  })
+
+  return createResponse
+}
+
+const getSubWalletToken = async (subWalletName) => {
+  const fetch = (await import('node-fetch')).default
```

Comment:
why we would need to `await` for an import? and it's not a global import?

just curious

Reply:
So this is an annoyance. At the moment our nodejs apps run in old school commonjs mode. That means we have `require` but not the `import`/`export` esm syntax. `fetch` v3 requires that you use esm (I honestly don't know why. I had three choices:
1. move to esm everywhere. This is what I did initially but it was getting out of hand. Long term this is the correct move
2. Use `fetch` v2
3. Dynamic import (what I chose). Note that `esm` `import` is not synchronous and hence the dyanmic import must be done with an await.